geo/src/algorithm/contains.rs

```diff
@@ -340,11 +340,29 @@ where
     T: CoordinateType,
 {
     fn contains(&self, coord: &Coordinate<T>) -> bool {
-        let sign_1 = utils::sign(coord, &self.0, &self.1);
-        let sign_2 = utils::sign(coord, &self.1, &self.2);
-        let sign_3 = utils::sign(coord, &self.2, &self.0);
+        let s = self.0.y * self.2.x - self.0.x * self.2.y
```

Comment:
Could you link to the SO solution and / or that this function uses barycentric coordinates to obtain the answer? Just so readers know what's happening and where to look for more information.